### PR TITLE
Update tag parser

### PR DESCRIPTION
**Pull request by ruihan00**
## Status
Completed

## Description
Update tag parsing to check for predefined tag
Limit tags to 1 and change the format of the json data file

Expenditure - Food, Groceries, Entertainment, Housing , Shopping, Transport, Education, Housing, Others
Income - Salary, Allowance, Profits, Investment, Gifts, Others

## Checklist
- [x] update tag parser
- [x] limit number of tags to 1
- [x]  testing and debugging

### CODE REVIEW

**Review by JasonYapzx — COMMENTED** [review]
Have not tried the code but from looking through it looks good! @sharmaine1028 the intended tag is one per entry right, as per the implementation here.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -103,26 +100,33 @@ public static Date parseDate(String date) throws ParseException {
      *
      * @throws ParseException if the given {@code tag} is invalid.
      */
-    public static Tag parseTag(String tag) throws ParseException {
+    public static Tag parseTag(EntryType type, String tag) throws ParseException {
         requireNonNull(tag);
         String trimmedTag = tag.trim();
-        if (!Tag.isValidTagName(trimmedTag)) {
-            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
+        if (!Tag.isValidTagName(type, trimmedTag)) {
```
Comment (sharmaine1028) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:106]: The code here seems a little confusing, if the tag is invalid, then proceed to switch case?

Comment (sharmaine1028) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:106]: Perhaps it would be better to add a guard clause instead?

**Review by sharmaine1028 — COMMENTED** [review]
Good job with the creating the income and expenses tags
Just a minor thing to note regarding the isValidTagName function

**Comment by sharmaine1028** [comment]
> Have not tried the code but from looking through it looks good! @sharmaine1028 the intended tag is one per entry right, as per the implementation here.

Yep, each entry should be assigned one compulsory tag

**Review by sharmaine1028 — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (sharmaine1028) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:106]: Actually, upon looking closer, I realised I made a mistake and you're already using a guard clause! Perhaps we could abstract away the exception handling part to a new function like getErrorMessagebyType(....) to make the code more readable? But it's just a suggestion, considering the code here is already clear in what it does 👍

**Pull request merged**